## Libraries importieren

In [5]:
import pandas as pd

## Daten einlesen

Die Daten sind als csv-Daten im gleichen Ordner abgelegt, versehen mit dem Datum wann die Seite gescrapt wurde.

In [13]:
path = "anzahl_jobs_04-11-2018.csv"

with open(path) as f:
    dateiinhalt = f.read()

In [14]:
print(dateiinhalt)

# @ THIERRY: Hier fängt mein Problem an. ich will eine tabelle. eine Variable sind Berufsbezeichnungen, das andere das Resultat
# So wie das daher kommt komme ich unten nich weiter! zB alle Resultate zusammenzählen.

Abdichter/in EFZ,Abdichtungspraktiker/in EBA,Agrarpraktiker/in EBA,Anlagen- und Apparatebauer/in EFZ,Anlagenführer/in EFZ,Architekturmodellbauer/in EFZ,Assistent/in Gesundheit und Soziales EBA,Augenoptiker/in EFZ,Automatiker/in EFZ,Automatikmonteur/in EFZ,Automobil-Assistent/in EBA,Automobil-Fachmann/-frau EFZ,Automobil-Mechatroniker/in EFZ,Bäcker/in-Konditor/in-Confiseur/in EBA,Bäcker/in-Konditor/in-Confiseur/in EFZ,Baumaschinenmechaniker/in EFZ,Baupraktiker/in EBA,Bauwerktrenner/in EFZ,Bekleidungsgestalter/in EFZ,Bekleidungsnäher/in EBA,Betonwerker/in EFZ,Boden-Parkettleger/in EFZ,Bootbauer/in EFZ,Bootfachwart/in EFZ,Buchhändler/in EFZ,Büchsenmacher/in EFZ,Bühnentänzer/in EFZ,Büroassistent/in EBA,Carrosserielackierer/in EFZ,Carrosseriespengler/in EFZ,Chemie- und Pharmapraktiker/in EBA,Chemie- und Pharmatechnologe/-login EFZ,Coiffeur/-euse EBA,Coiffeur/-euse EFZ,Dachdecker/in EFZ,Dachdeckerpraktiker/in EBA,Dekorationsnäher/in EBA,Dentalassistent/in EFZ,Detailhandelsassistent/in EBA,"D

In [10]:
# csv-Datei hat keine Kopfzeile. Diese Info muss beim Öffnen mitgegeben werden

df = pd.read_csv(path, header=None)

In [11]:
df

0                            1                      2    \
0  Abdichter/in EFZ  Abdichtungspraktiker/in EBA  Agrarpraktiker/in EBA   
1               135                           44                     57   

                                 3                     4    \
0  Anlagen- und Apparatebauer/in EFZ  Anlagenführer/in EFZ   
1                                209                   137   

                             5                                         6    \
0  Architekturmodellbauer/in EFZ  Assistent/in Gesundheit und Soziales EBA   
1                              0                                       478   

                   7                   8                        9    \
0  Augenoptiker/in EFZ  Automatiker/in EFZ  Automatikmonteur/in EFZ   
1                  122                 365                      103   

            ...                                    231  \
0           ...             Unterhaltspraktiker/in EBA   
1           ...                                     50   

                                232                            233  \
0  Veranstaltungsfachmann/-frau EFZ  Vergolder/in-Einrahmer/in EFZ   
1                                12                              1   

                                234                        235            236  \
0  Verpackungstechnologe/-login EFZ  Weintechnologe/-login EFZ  Winzer/in EFZ   
1                                16                         18             39   

                          237                   238              239  \
0  Wohntextilgestalter/in EFZ  Zahntechniker/in EFZ  Zeichner/in EFZ   
1                          11                    14              397   

                       240  
0  Zimmermann/Zimmerin EFZ  
1                      431  

[2 rows x 241 columns]

In [19]:
pd.melt(df)

variable                                      value
0           0                           Abdichter/in EFZ
1           0                                        135
2           1                Abdichtungspraktiker/in EBA
3           1                                         44
4           2                      Agrarpraktiker/in EBA
5           2                                         57
6           3          Anlagen- und Apparatebauer/in EFZ
7           3                                        209
8           4                       Anlagenführer/in EFZ
9           4                                        137
10          5              Architekturmodellbauer/in EFZ
11          5                                          0
12          6   Assistent/in Gesundheit und Soziales EBA
13          6                                        478
14          7                        Augenoptiker/in EFZ
15          7                                        122
16          8                         Automatiker/in EFZ
17          8                                        365
18          9                    Automatikmonteur/in EFZ
19          9                                        103
20         10                 Automobil-Assistent/in EBA
21         10                                         89
22         11               Automobil-Fachmann/-frau EFZ
23         11                                        607
24         12             Automobil-Mechatroniker/in EFZ
25         12                                        558
26         13     Bäcker/in-Konditor/in-Confiseur/in EBA
27         13                                         74
28         14     Bäcker/in-Konditor/in-Confiseur/in EFZ
29         14                                        338
..        ...                                        ...
452       226  Tiermedizinische/r Praxisassistent/in EFZ
453       226                                         23
454       227                         Tierpfleger/in EFZ
455       227                                         20
456       228                       Uhrenarbeiter/in EBA
457       228                                         26
458       229                           Uhrmacher/in EFZ
459       229                                         46
460       230                Uhrmacher/in Produktion EFZ
461       230                                         21
462       231                 Unterhaltspraktiker/in EBA
463       231                                         50
464       232           Veranstaltungsfachmann/-frau EFZ
465       232                                         12
466       233              Vergolder/in-Einrahmer/in EFZ
467       233                                          1
468       234           Verpackungstechnologe/-login EFZ
469       234                                         16
470       235                  Weintechnologe/-login EFZ
471       235                                         18
472       236                              Winzer/in EFZ
473       236                                         39
474       237                 Wohntextilgestalter/in EFZ
475       237                                         11
476       238                       Zahntechniker/in EFZ
477       238                                         14
478       239                            Zeichner/in EFZ
479       239                                        397
480       240                    Zimmermann/Zimmerin EFZ
481       240                                        431

[482 rows x 2 columns]

In [12]:
df.dtypes

0      object
1      object
2      object
3      object
4      object
5      object
6      object
7      object
8      object
9      object
10     object
11     object
12     object
13     object
14     object
15     object
16     object
17     object
18     object
19     object
20     object
21     object
22     object
23     object
24     object
25     object
26     object
27     object
28     object
29     object
        ...  
211    object
212    object
213    object
214    object
215    object
216    object
217    object
218    object
219    object
220    object
221    object
222    object
223    object
224    object
225    object
226    object
227    object
228    object
229    object
230    object
231    object
232    object
233    object
234    object
235    object
236    object
237    object
238    object
239    object
240    object
Length: 241, dtype: object

In [7]:
df.shape

(1, 241)

**RESULTAT 1:** Im Lehrstellennachweis LENA gibt es **241** verschiedene Berufe.

In [15]:
df[value].count

NameError: name 'value' is not defined